src/transactions/financial.ts

```diff
@@ -263,3 +309,61 @@ export const constructTransferCreatorCoinTransaction = (
     }
   );
 };
+
+const buildBuyCreatorCoinMetadata = (params: BuyCreatorCoinRequestParams) => {
+  const metadata = new TransactionMetadataCreatorCoin();
+  metadata.profilePublicKey = bs58PublicKeyToCompressedBytes(
+    params.CreatorPublicKeyBase58Check
+  );
+  metadata.operationType = 0;
+  metadata.desoToSellNanos = params.DeSoToSellNanos;
+  metadata.desoToAddNanos = params.DeSoToAddNanos ?? 0;
```

Comment:
This will always be 0 for both CC buys and sells. It's not ever used for anything actually.